[Package]: Pounce

### Package description

pounce(1) is a multi-client, TLS-only IRC bouncer. It maintains a persistent connection to an IRC server, acting as a proxy and buffer for a number of clients. When a client connects, any messages received since it last disconnected will be relayed to it. Unlike some other bouncers, pounce uses a single buffer for all IRC messages, which acts as a queue from which each client reads messages independently.

### Home page URL

https://git.causal.agency/pounce/about/

### Source code URL

https://git.causal.agency/pounce

### Packaging policy acknowledgement

- [X] I have read and understand the [Packaging Policy](https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md#packaging-policy).

### Additional information

Since libretls is merged its Time for pounce now https://github.com/termux/termux-packages/pull/9623

As a note, pounce bouncer is not available at several distribution packages manager. Also it requires some patch so android could support it.

```
cc  bounce.o client.o config.o local.o ring.o server.o state.o xdg.o -lcrypt -L/data/data/com.termux/files/usr/lib -ltls -o pounce
ld.lld: error: undefined symbol: explicit_bzero
>>> referenced by bounce.c
>>>               bounce.o:(main)
>>> referenced by bounce.c
>>>               bounce.o:(main)
>>> referenced by client.c
>>>               client.o:(handlePass)
>>> referenced 2 more times
clang-13: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Makefile:39: pounce] Error 1
```

--- a/packages/pounce/bounce.h.patch
+++ b/packages/pounce/bounce.h.patch
@@ -1,0 +1,17 @@
+--- a/bounce.h
++++ b/bounce.h
+@@ -34,6 +34,14 @@
+ #include <sys/time.h>
+ #include <tls.h>
+ 
++#ifdef __ANDROID__
++#include <crypt.h>
++
++static void *(*volatile explicit_memset)(void *, int, size_t) = memset;
++#undef explicit_bzero
++#define explicit_bzero(s, n) explicit_memset(s, 0, n)
++#endif
++
+ #ifndef OPENSSL_BIN
+ #define OPENSSL_BIN "openssl"
+ #endif

--- a/packages/pounce/build.sh
+++ b/packages/pounce/build.sh
@@ -1,0 +1,13 @@
+TERMUX_PKG_HOMEPAGE=https://git.causal.agency/pounce
+TERMUX_PKG_DESCRIPTION="A multi-client, TLS-only IRC bouncer"
+TERMUX_PKG_LICENSE="GPL-3.0"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=3.0
+TERMUX_PKG_SRCURL=https://git.causal.agency/pounce/snapshot/pounce-${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=f776f7f170493697a97923e7dce9597dff5577fd40ba756e9a1bcfab17199df0
+TERMUX_PKG_DEPENDS="libcrypt, libretls"
+TERMUX_PKG_BUILD_IN_SRC=true
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
+--mandir=$TERMUX_PREFIX/share/man
+"
+TERMUX_PKG_EXTRA_MAKE_ARGS="all"

--- a/packages/pounce/pounce.1.patch
+++ b/packages/pounce/pounce.1.patch
@@ -1,0 +1,11 @@
+--- a/pounce.1
++++ b/pounce.1
+@@ -627,7 +627,7 @@
+ followed by the colon-separated list of paths
+ .Ev $XDG_CONFIG_DIRS ,
+ usually
+-.Pa /etc/xdg .
++.Pa @TERMUX_PREFIX@/etc/xdg .
+ .It Pa ~/.config/pounce
+ The most likely location of configuration files.
+ .

--- a/packages/pounce/xdg.c.patch
+++ b/packages/pounce/xdg.c.patch
@@ -1,0 +1,19 @@
+--- a/xdg.c
++++ b/xdg.c
+@@ -49,14 +49,14 @@
+ 	.envHome = "XDG_CONFIG_HOME",
+ 	.envDirs = "XDG_CONFIG_DIRS",
+ 	.defHome = ".config",
+-	.defDirs = "/etc/xdg",
++	.defDirs = "@TERMUX_PREFIX@/etc/xdg",
+ };
+ 
+ static const struct Base Data = {
+ 	.envHome = "XDG_DATA_HOME",
+ 	.envDirs = "XDG_DATA_DIRS",
+ 	.defHome = ".local/share",
+-	.defDirs = "/usr/local/share:/usr/share",
++	.defDirs = "@TERMUX_PREFIX@/local/share:@TERMUX_PREFIX@/share",
+ };
+ 
+ static char *basePath(

